lib/src/orchestration.dart

```diff
@@ -57,24 +70,29 @@ Future<Leaks> withLeakTracking(
     ),
   );
 
-  await callback();
-
-  await _forceGC(
-    gcCycles: gcCountBuffer,
-    timeout: timeoutForFinalGarbageCollection,
-  );
-
-  final result = collectLeaks();
-
   try {
-    if (result.total > 0 && throwOnLeaks) {
-      throw MemoryLeaksDetectedError(result);
+    await callback();
+
+    asyncCodeRunner ??= (action) => action();
+    await asyncCodeRunner(
+      () async => await _forceGC(
+        gcCycles: gcCountBuffer,
+        timeout: timeoutForFinalGarbageCollection,
+      ),
+    );
+
+    final leaks = collectLeaks();
+
+    if (leaks.total > 0 && shouldThrowOnLeaks) {
+      // `expect` should not be used here, because, when the method is used
+      // from Flutter, the packages `test` and `flutter_test` conflict.
+      MemoryLeaksDetectedError(leaks);
```

Comment:
```suggestion
      throw MemoryLeaksDetectedError(leaks);
```
Does this need to be thrown?